rust/main/lander/src/tests/evm/tests_inclusion_stage.rs

```diff
@@ -1099,6 +1103,121 @@ async fn test_tx_finalized_but_failed() {
     );
 }
 
+/// Test case where tx is assigned a new nonce, but tx already exists.
+/// We still want to make sure nonce is reflected in the tx.
+#[tokio::test]
+#[traced_test]
+async fn test_tx_new_nonce_but_tx_already_exists() {
```

Comment:
We need a better description and better test method name to make it clear what is happening in this test case.